Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java

```diff
@@ -277,40 +326,61 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+    private ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target,
         boolean isRequesting, Properties webContextProperties, String serviceName) {
+        ActiveParticipant participant = new ActiveParticipant();
+        String hostAddress = null;
 
-        String strUrl;
-        String strHost;
-
-        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        if (serviceName.equals(NhincConstants.DOC_RETRIEVE_SERVICE_NAME) && isRequesting) {
+            hostAddress = getLocalHostAddress();
+            participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
+            participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+            participant.setNetworkAccessPointID(hostAddress);
+            participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(hostAddress));
 
-        strUrl = isRequesting ? getWebServiceUrlFromRemoteObject(target, serviceName)
-            : getWebServiceRequestUrl(webContextProperties);
-        if (strUrl != null) {
-            try {
-                URL url = new URL(strUrl);
-                participant.setUserID(strUrl);
-                strHost = url.getHost();
-                participant.setNetworkAccessPointID(strHost);
-                participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(strHost));
-            } catch (MalformedURLException ex) {
-                LOG.error(ex);
-                // The url is null or not a valid url; for now, set the user id to anonymous
-                participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
-                // TODO: For now, hardcode the value to localhost; need to find out if this needs to be set
-                participant.setNetworkAccessPointTypeCode(AuditTransformsConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_NAME);
-                participant.setNetworkAccessPointID(AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS);
+        } else {
+            if ((serviceName.equals(NhincConstants.DOC_RETRIEVE_SERVICE_NAME)) && !isRequesting) {
+                hostAddress = getLocalAddressFromProperties(webContextProperties);
+            } else {
+                hostAddress = isRequesting ? getWebServiceUrlFromRemoteObject(target, serviceName)
+                    : getWebServiceRequestUrl(webContextProperties);
+            }
+            if (hostAddress != null) {
+                try {
+                    URL url = new URL(hostAddress);
+                    participant.setUserID(hostAddress);
+                    hostAddress = url.getHost();
+                    participant.setNetworkAccessPointID(hostAddress);
+                    participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(hostAddress));
+                } catch (MalformedURLException ex) {
+                    LOG.error(ex);
```

Comment:
When logging an exception, please use the following format:

LOG.<level>("<Basic description of problem>: " + ex.getLocalizedMessage(), ex);